docs/Hardening.md

```diff
@@ -0,0 +1,197 @@
+# Linux Hardening while using OMS Log Collector
+We support some industry standard hardening (please see [Supported Hardening or Customization](#supported-hardening-or-customization) for more details).
+We also can provide some [troubleshooting steps](#common-hardening-issues) for customized hardening.
+
+# Table of Contents
+- [Supported Hardening or Customization](#supported-hardening-or-customization)
+- [Common Hardening Issues](#common-hardening-issues)
+- [Documentation](#documentation)
+
+## Supported Hardening or Customization
+### Hardening with FIPS
+FIPS hardening is supported in OMS versions 1.13 and newer.
+
+How to check if FIPS is enabled:
+```
+$> cat /proc/sys/crypto/fips_enabled
+1
+$> sysctl crypto.fips_enabled
+crypto.fips_enabled = 1
+```
+
+### Hardening with Cylance Anti-Virus
+Cylance Anti-Virus hardening is supported in OMS versions 1.12 and newer.
+
+### Not Yet Supported Hardening
+- SELinux
+- Center for Internet Security (CIS)
+
+## Common Hardening Issues
+If you're using a customized hardening setup, and are having issues running the OMS Agent, try going through these common problems encountered with hardening.
+### Directory Permissions
+#### Issue
+Several major folders in Linux need to have appropriate file permissions, otherwise the OMS Agent won't run properly. Specifically, we need the following:
+| Directory | Permissions (numeric value) | Permissions (symbolic notation) |
+| --- | --- | --- |
+| `/var` | 755 | `drwxr-xr-x` |
+| `/var/log` | 775 | `drwxrwxr-x` |
+| `/etc` | 775 | `drwxrwxr-x` |
+| `/tmp` | 775 | `drwxrwxr-x` |
+
+#### How to Check
+You can check what the permissions for these folders are by the following:
+```
+$> ls -l / | grep var
```

Comment:
is there a way that we could do this in a single command to simplify this for customers? e.g. (untested) `grep -E '/var|/var/log|...'`

Reply:
I couldn't get it in one command (since /var/log isn't at the same level as the rest), but I can combine two commands in one line to get everything, do you think the below is okay?
```
$> ls -l / | grep -E 'var|etc|opt|tmp'; ls -ld /var/log
drwxr-xr-x.  85 root root 8192 Nov 13 00:55 etc
drwxr-xr-x.   6 root root   55 Nov 10 18:15 opt
drwxrwxrwt.   7 root root   93 Nov 18 03:13 tmp
drwxr-xr-x.  20 root root  282 Nov 13 00:54 var
drwxr-xr-x. 10 root root 4096 Nov 15 03:36 /var/log
```

Reply:
Changed to one line command:
https://github.com/microsoft/OMS-Agent-for-Linux/pull/1269/commits/b6a96c6bef0fd670a3c6676c17026d5228135af5#diff-ff8800189743d6567bcda8a83de5b22c943b28c51e4488cce0ea2b185716cc37R44